### DISCUSSION TITLE
Error al insertar gráfico

### DISCUSSION THREAD

**Discussion by EAleman8** [LaunchX-InnovaccionVirtual/CursoIntroPython#111] [Q&A] [answered]
Hola!
Estaba siguiendo los ejemplos de la documentación del modulo 0, pero tengo problemas al ejecutar el código para insertar el gráfico de los niveles de oxígeno, me muestra este error.

![imagen](https://user-images.githubusercontent.com/99446604/153533797-2c8e7621-eab9-4b8b-9daf-a1f1dfd6f910.png)

Alguien sabe a que se debe este problema?
P.D. Estoy tratando de acceder a los lives de la sesion 1 y 2 de python porque no los vi, pero ya no me cargan, se queda la pantalla en negro, alguien más está teniendo ese problema?

**Comment by JulioCampos94** [comment]
Hola, trata de hacer la instalación de los pip sin el "%"
pip install matplotlib
pip install numpy

**Reply by EAleman8** [reply]
Ya lo intenté de esa forma y me arroja error.

![imagen](https://user-images.githubusercontent.com/99446604/153534657-88e28809-6997-4374-b10d-bc8f9933c2df.png)

Esos pip solo pude instalarlos con el %, no me los instala de ninguna otra forma, ni con el !

**Reply by Juan-Cruz15** [reply]
Yo lo intente de todas las formas que han dicho y no he podido solucionarlo 
ya pase a versión 3.9.10 
intente ponerlo tal cual viene en el documento 
intente con el % 
y con --upgrade
alguien me puede apoyar por favor 
![screenshot_20220210_215512](https://user-images.githubusercontent.com/99099757/153535333-1248a0e3-2746-4fb4-a63c-7298666486ff.png)

**Reply by EAleman8** [reply]
Hola!
Veo que está colocada la instalación del numpy, pero no la del matplotlib, ingresa el código para instalar ambas:

%pip install matplotlib
%pip install numpy

Y después corre el código de los niveles de oxígeno tal como lo indica JulioCampos94, colocando todas las lineas de código a la misma altura:

![imagen](https://user-images.githubusercontent.com/99446604/153535877-e1cb4682-69f5-419c-b2fa-65fef7880f0d.png)

**Reply by Juan-Cruz15** [reply]
Muchas gracias, ya quedo solucionado :)

**Reply by IrvinSoto** [reply]
Gracias @EAleman8 también me ayudaste a mi :D pero... quisiera saber cual es el motivo de porque no las instala con el signo de admiración como se indica en la guía.

me sirvio con % pero no con !

**Reply by JulsJuarez** [reply]
Hola, creo que el error es que al copiar, el signo de ! Se recorre 

Los 2 !pip deben estar al inicio

**Comment by JulioCampos94** [comment, marked as answer]
ok, no habia notado que el error de la imagen te lo manda por la identación.
En python las identaciones de las lineas son motivo para que no compile, entonces todo tu codigo debe estar a la misma altura que los import.

import numpy as np
import matplotlib as plt
data = np......
oxy_nums = ...

**Reply by EAleman8** [reply]
Perfecto!
Quedó solucionado, muchas gracias!
